runbotics/runbotics-orchestrator-ui/src/src-app/hooks/useProcessActionVariables.ts

```diff
@@ -183,16 +183,21 @@ const useProcessActionVariables = (selectedElementParentId?: string) => {
             .filter((item: ActionVariableObject[]) => item.length > 0)
             .flat();
 
-        if(!selectedElementParentId) return { inputActionVariables: inputActionVars, outputActionVariables: outputActionVars };
+        const loopIds = allVarsAssigningActions
+            .filter((element: ScopedModdleElement) => element.actionId === 'loop.loop')
+            .map((element: ScopedModdleElement) => element.id);
+
+        const loopIterator = loopIds.includes(selectedElementParentId) ? [{ name: 'iterator', value: 'variable' }] : [];
```

Comment:
does it work if you have loop inside another loop? If yes, won't they be both "#{iterator}"?

it may be hard to differ them - should we suffix them with e.g. scopeId? like #{iterator-s2kfw43e}? I don't think so - still don't know which loop is which.
Maybe we should give access only to the closest scope iterator (for case when loop is inside another loop)? - i suggest to discuss it with Mikołaj.

Reply:
Each loop recognizes its own iterator, so even in nested loops it works perfectly. I've also updated documentation, how to deal with nested loops using variables.